src/amo/reducers/addonsByAuthors.js

```diff
@@ -0,0 +1,120 @@
+/* @flow */
+import type { AddonType } from 'core/types/addons';
+
+
+type State = {
+  byAddon: { [string]: AddonType },
+};
+
+export const initialState: State = {
+  byAddon: {},
+};
+
+export const NB_OTHERS_ADDONS_BY_AUTHORS = 6;
+
+export const FETCH_OTHER_ADDONS_BY_AUTHORS = 'FETCH_OTHER_ADDONS_BY_AUTHORS';
+const LOAD_OTHER_ADDONS_BY_AUTHORS = 'LOAD_OTHER_ADDONS_BY_AUTHORS';
+
+type FetchOtherAddonsByAuthorsParams = {|
+  errorHandlerId: string,
+  addonType: string,
+  authors: Array<string>,
+  slug: string,
+|};
+
+type FetchOtherAddonsByAuthorsAction = {|
+  type: 'FETCH_OTHER_ADDONS_BY_AUTHORS',
```

Comment:
It's tricky for Flow to figure out the `const` value because any dynamic JS expression can be used to set the value of the `const`.

However, there is a way to at least keep the repetition in one place:

```js
export const 
  // This declares the `const` as a literal type, not a string type.
  FETCH_OTHER_ADDONS_BY_AUTHORS: 'FETCH_OTHER_ADDONS_BY_AUTHORS'
  // This sets the actual JS value.
  = 'FETCH_OTHER_ADDONS_BY_AUTHORS';

type FetchOtherAddonsByAuthorsAction = {|
  // Now you can reference its type.
  type: typeof FETCH_OTHER_ADDONS_BY_AUTHORS,
|}
```

However, if we had full Flow coverage then I don't see why we would need to define a constant at all. We could just use the string and Flow would warn us when values did not match up. The main reason we use `const` values now is because the linter will warn us about invalid imports. Since we won't reach full Flow coverage we should probably keep doing that.

Reply:
Here's another approach of creating a typed reducer without repetition: https://gist.github.com/rendro/4c9f4ddea8133bbddc3df427a8f62b9e